tests/pytest/hf/test_uhf.py

```diff
@@ -0,0 +1,66 @@
+"""Test the HF solver."""
+
+import pytest
+
+from forte import Molecule, Basis
+from forte.solvers import solver_factory, HF
+
+
+def test_uhf():
+    """Test UHF computation of the B1 state of methylene."""
+
+    ref_energy = -38.92655952236967
+
+    # create a molecule from a string
+    mol = Molecule.from_geom("""
+    C
+    H 1 1.085
+    H 1 1.085 2 135.5
+    """)
+
+    # create a basis object
+    basis = Basis('cc-pVDZ')
+
+    # create a molecular model
+    root = solver_factory(molecule=mol, basis=basis)
+
+    # specify the electronic state
+    state = root.state(charge=0, multiplicity=3, sym='b1')
+
+    # define a HF object
+    hf = HF(root, state=state, restricted=False)
+    hf.run()
+
+    assert hf.value('hf energy') == pytest.approx(ref_energy, 1.0e-10)
+
+
+def test_uhf_wrong_sym():
+    """Test UHF computation that lands on the wrong symmetry state."""
+
+    # create a molecule from a string
+    mol = Molecule.from_geom("""
+    C
+    H 1 1.085
+    H 1 1.085 2 135.5
+    """)
+
+    # create a basis object
+    basis = Basis('cc-pVDZ')
+
+    # create a molecular model
+    root = solver_factory(molecule=mol, basis=basis)
+
+    # specify the electronic state
+    state = root.state(charge=0, multiplicity=3, sym='a1')
+
+    # define a HF object
+    hf = HF(root, state=state, docc=[2, 0, 0, 1], socc=[1, 0, 1, 0], restricted=False)
+
+    # should raise an error since we get a state with different symmetry
+    with pytest.raises(RuntimeError):
```

Comment:
Do we want to run validation when the HF object is created, or when you call its `run` method? The inconsistency with the occupation and the target state is clear from the arguments.

Reply:
This is a good question. The symmetry cannot be checked at creation, since without a docc/socc, the psi4 HF algorithm is allowed to identify the best solution (but is not aware of a target symmetry) and could end up choosing a state with the wrong symmetry. Ideally, one day we could write a simple HF code that can converge on a desired symmetry. An alternative would be to ask the user to always specify the docc/socc, but that would still make the HF computation prone to error. Unfortunately no easy answer here!